### PR TITLE
CASMINST-5257: Add security hardening guide to CSM 1.3, insert into i…

### PR DESCRIPTION
**Pull request by jeremy-duckworth**
# Description

* Add security hardening guide for 1.3 release. Include in install and upgrade flows.
* Include xname-filtering, iPXE filename customization in hardening guide.
* Move Kubernetes API Encryption to security hardening guide (optional step). 

# Checklist Before Merging

- [X] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

### CODE REVIEW

**Comment by jeremy-duckworth** [comment]
+ @kburns-hpe this PR adds the setup for xname filtering into security hardening guide as optional procedure.

**Comment by jeremy-duckworth** [comment]
@mitcharf  this PR moves Kubernetes API Encryption setup into the security hardening guide as an optional procedure.

File: install/README.md [hunk 1]
```diff
@@ -203,9 +203,10 @@ for additional information on system hardware, troubleshooting, and administrati
 
 See [Troubleshooting Installation Problems](troubleshooting_installation.md).
 
-### 11. Kubernetes encryption
+## Post-installation
 
-As an optional post installation task, encryption of Kubernetes secrets may be enabled. This enables
-at rest encryption of data in the `etcd` database used by Kubernetes.
+### 11. Apply Security Hardening
```
Comment (manderson-hpe) [inline install/README.md:208]: Should be `Apple security hardening` to match the rest of the document.

File: operations/CSM_product_management/Apply_Security_Hardening.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+   Failure to apply the referenced configuration could result in NCN disk space exhaustion on Kubernetes Master Nodes.
+
+3. Customize ("randomize") iPXE Binary Name.
+
+   Perform procedure(s) in [Customize iPXE Binary Names](../boot_orchestration/Customize_iPXE_Binary_Names.md).
+
+4. (Optional) Enable Spire and OPA Xname Validation.
+
+    Perform procedure(s) in [Xname Validation](../spire/xname_validation.md).
```
Comment (rustydb) [inline operations/CSM_product_management/Apply_Security_Hardening.md:29]: nit: use lowercase "xname"

https://github.com/Cray-HPE/docs-csm/blob/main/.spelling#L621-L622

```suggestion
    Perform procedure(s) in [xname Validation](../spire/xname_validation.md).
```

File: operations/CSM_product_management/Apply_Security_Hardening.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+   If Kubernetes API Auditing was enabled at install, perform procedure(s) in [Limit Kubernetes API Audit Log Maximum Backups](../kubernetes/Limit_Kubernetes_API_Audit_Log_Maxbackups.md).
+
+   Failure to apply the referenced configuration could result in NCN disk space exhaustion on Kubernetes Master Nodes.
+
+3. Customize ("randomize") iPXE Binary Name.
+
+   Perform procedure(s) in [Customize iPXE Binary Names](../boot_orchestration/Customize_iPXE_Binary_Names.md).
+
+4. (Optional) Enable Spire and OPA Xname Validation.
```
Comment (rustydb) [inline operations/CSM_product_management/Apply_Security_Hardening.md:27]: nit: use lowercase "xname"

https://github.com/Cray-HPE/docs-csm/blob/main/.spelling#L621-L622

```suggestion
4. (Optional) Enable Spire and OPA xname Validation.
```

**Review by rustydb — CHANGES_REQUESTED** [review]

File: upgrade/README.md [hunk 4]
```diff
@@ -99,7 +100,13 @@ The upgrade is a guided process starting with [Upgrade Management Nodes and CSM
         cray artifacts create config-data "${TARFILE}" "/root/${TARFILE}"
         ```
 
-## 5. Next topic
+## 5. Apply Security Hardening
```
Comment (manderson-hpe) [inline upgrade/README.md:103]: Should also be `Apply security hardening`.

**Review by heemstra — APPROVED** [review]

**Pull request merged**